src/net.cpp

```diff
@@ -2638,3 +2612,17 @@ bool CBanDB::Read(banmap_t& banSet)
 int64_t PoissonNextSend(int64_t nNow, int average_interval_seconds) {
     return nNow + (int64_t)(log1p(GetRand(1ULL << 48) * -0.0000000000000035527136788 /* -1/2^48 */) * average_interval_seconds * -1000000.0 + 0.5);
 }
+
+/* static */ uint64_t CNode::CalculateKeyedNetGroup(const CAddress& ad)
+{
+    static uint64_t k0 = 0, k1 = 0;
```

Comment:
This code is not thread-safe - does it matter?

Otherwise it'd be better to make this a static instance of a structure with the initialization code in the constructor, and [C++11 semantics](http://stackoverflow.com/questions/8102125/is-local-static-variable-initialization-thread-safe-in-c11)  will make sure it will only get initialized once in a thread-safe way.